### ISSUE TITLE
Changing between soldier and sniper runtimes doesn't take effect until Steam is restarted

### ISSUE THREAD

**Issue by smcv** [ValveSoftware/steam-for-linux#9835]
#### Your system information

* Steam client version (build number or date): beta 1689034492, 2023-07-11
* Distribution (e.g. Ubuntu): Debian 12, also reproduced on Ubuntu 22.04
* Opted into Steam client beta?: yes
* Have you checked for system updates?: yes
* Steam Logs: [logs from Debian 12 system, without attempting any workarounds](https://github.com/ValveSoftware/steam-for-linux/files/12029072/steam-logs.tar.gz)
* GPU: AMD Vega, Mesa 22.3.6-1+deb12u1; also reproduced on NVIDIA 1080 with proprietary driver

#### Please describe your issue in as much detail as possible:

When the configuration for a new version of a game changes to run in a different Steam Linux Runtime by default, Steam does not immediately respond to that change, and instead continues to run it with the old configuration. The result is that the new game runs in the old runtime, which is not expected to work, and can lead to crashing or other brokenness ([example](https://github.com/ValveSoftware/Dota-2/issues/2390)).

#### Steps for reproducing this issue:

1. Have Dota 2 already installed, at a version **older than** the recent update that switched it from "Steam Linux Runtime" (scout layered onto soldier) to Steam Linux Runtime 3 (sniper)
2. Run Steam in a terminal, or otherwise in some way that lets you see its logging output
3. Let Steam update Dota 2 to the latest version
4. **Don't** restart Steam
5. Try to launch Dota 2
6. Look at what is logged by Steam

What I saw was this command-line (paths abbreviated and newlines added here for readability):

```
.../reaper SteamLaunch AppId=570 --
.../steam-launch-wrapper --
.../SteamLinuxRuntime_soldier/_v2-entry-point --verb=waitforexitandrun --
.../SteamLinuxRuntime/scout-on-soldier-entry-point-v2 --
.../dota 2 beta/game/dota.sh +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0
```

Notice that it says `soldier` and `scout-on-soldier`, which is incorrect for the new version of Dota 2.

#### Workaround

On the Ubuntu 22.04 system where I also reproduced this, I was able to work around this as follows:

1. (as above)
2. (as above)
3. Let Steam update Dota 2 to the latest version, as above
4. Completely exit from Steam
5. Run Steam again
6. Try to launch Dota 2 again

This resulted in me seeing the correct command-line, something like this:

```
.../reaper SteamLaunch AppId=570 --
.../steam-launch-wrapper --
.../SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun --
.../dota 2 beta/game/dota.sh +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0
```

Notice that this time, it says `sniper`, which is correct for the new version of Dota 2.

**Comment by smcv** [comment]
Because this is a transient situation, I will be unable to reproduce it after today, but hopefully Valve developers will be able to reproduce it on-demand by swapping the runtime used for a non-public test title.

**Comment by smcv** [comment]
We saw similar issues when "Proton Experimental" changed from being 7.0-based (running on soldier) to 8.0-based (running on sniper). Those are believed to have been fixed already (at least in the beta client), but it seems those fixes were not comprehensive.

**Comment by smcv** [comment]
After opening the developer console with `steam steam://open/console`, I can enter `app_info_print 570 4` and the resulting metadata correctly contains `SteamLinuxRuntime_sniper`. But the game still tries to launch in soldier, so that updated information is not getting through to the part of Steam that runs games.

**Comment by smcv** [comment]
You should not need to make any symbolic links between SteamLinuxRuntime and Dota 2: that is not how any of these runtimes are meant to work. Please don't encourage others to use that workaround.

As I said above, the recommended workaround for this is: after letting Steam update Dota 2, completely exit from Steam, then re-run it. It will hopefully download "Steam Linux Runtime - sniper" into your Steam library automatically, either immediately or when you try to run Dota 2 (but I cannot verify that, because I already had sniper installed). If not, running `steam steam://install/1628350` will force sniper to be installed.

After that, launching Dota 2 should work correctly (it will launch via sniper). If not, please report a separate issue with full details.

**Comment by sylware** [comment]
Okay, I'll try to remove the symbolic links and the environment variable for libthai (without LIBTHAI_DICTDIR or /usr/share/libthai with its data, libthai will crash dota2 in many places).

Hopefully, LIBTHAI_DICTDIR is properly setup before starting dota2 (and I suspect similar issue with ALSA_CONFIG_DIR and XLOCALEDIR).

**Comment by sylware** [comment]
@smcv Nope, same issue. Then I manually installed the "sniper" steamruntime 1628350.
In logs/compat_log.txt I have clearly:
Requesting mapping AppID 570 from appinfo to tool "SteamLinuxRuntime_sniper"
 Mapping AppID 570 to tool "SteamLinuxRuntime_sniper" with priority 85
...
Registering tool steamlinuxruntime_sniper, AppID 1628350
Loaded manifest for tool 1628350.

But nope, dota2 is not run in the "sniper" steamruntime.

**Comment by manticore-projects** [comment]
> If not, running `steam steam://install/1628350` will force sniper to be installed.
> 
> After that, launching Dota 2 should work correctly (it will launch via sniper). If not, please report a separate issue with full details.

Greetings.
I have tried exactly that and "Sniper" has been downloaded. However, Dota still does not work:

```
Setting breakpad minidump AppID = 373300
07/13 06:29:51 SetAppID Setting app ID to 373300.
Loaded libSDL2-2.0.so.0, got 0x5640002944e0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x563ffff3ca30
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x5640002e1440
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x564000814810
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x564000b6f480
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x564000b68520
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x564000b46f80
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x564000b5d3e0
Fossilize INFO: Overriding serialization path: "/home/are/.local/share/Steam/steamapps/shadercache/570/fozpipelinesv6/steamapprun_pipeline_cache".
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libmaterialsystem2.so, got 0x5640042865e0
Loaded libpangoft2-1.0.so, got (nil)
 failed to dlopen "libpangoft2-1.0.so" error=/usr/lib/libharfbuzz.so.0: undefined symbol: FT_Get_Color_Glyph_Layer
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so, got (nil)
 failed to dlopen "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so" error=/usr/lib/libharfbuzz.so.0: undefined symbol: FT_Get_Color_Glyph_Layer
```

**Comment by Espionage724** [comment]
Can Dota 2 not be ran with system libraries anymore? I was just running Dota 2 for over a week with steamcmd and Goldberg after that context menu mess made Steam unusable, and [all I needed was a symlink for libbz2.so.1.0 to run Dota 2 directly](https://wiki.realmofespionage.xyz/games:dota_2).

Now I get this:

```
[espionage724@Spinesnap ~]$ cd ~/'Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64' && SDL_AUDIODRIVER='pipewire' SDL_VIDEODRIVER='x11' ~/'Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/dota2'
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x55d6d8fcbfe0
Using breakpad crash handler
[S_API] SteamAPI_Init(): SteamAPI_IsSteamRunning() did not locate a running instance of Steam.
[S_API] SteamAPI_Init(): Loaded '/home/espionage724/.steam/sdk64/steamclient.so' OK.
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561199874894657 [API loaded yes]
SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561199874894657
Setting breakpad minidump AppID = 373300
Loaded libSDL2-2.0.so.0, got 0x55d6d904c590
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x55d6d8f3a0a0
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x55d6d909c630
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x55d6d95ce4d0
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x55d6d9910a60
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x55d6d9906630
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x55d6d98ed180
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x55d6d9909120
ATTENTION: default value of option mesa_glthread overridden by environment.
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libmaterialsystem2.so, got 0x55d6dbd7b900
Loaded libpangoft2-1.0.so, got (nil)
 failed to dlopen "libpangoft2-1.0.so" error=libpangoft2-1.0.so: cannot open shared object file: No such file or directory
Loaded /home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so, got (nil)
 failed to dlopen "/home/espionage724/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so" error=libbz2.so.1.0: cannot open shared object file: No such file or directory
Segmentation fault (core dumped)
```

Which doesn't make sense to me since `libbz2.so.1.0` is still in the folder, still symlinked, and it was just working fine this morning before I decided to do an app "update" on 570.

**Comment by smcv** [comment]
@sylware:
> But nope, dota2 is not run in the "sniper" steamruntime

Please get the output that Steam produces when you run Steam from a terminal, and run Dota 2 from there. There should be lines of output that look like this:

```
/bin/sh\0-c\0 <SOMETHING>
Game process added : AppID 570 "<SOMETHING>"
chdir "<SOME DIRECTORY>"
```

I want to see what replaces `<SOMETHING>` in those lines. I expect that the first two should be very similar. Please copy/paste the exact text from the terminal rather than paraphrasing it: the specifics matter.

If it contains `steamapps/common/SteamLinuxRuntime_sniper/'_v2-entry-point` then whatever problem you are having is not this bug, and should be discussed elsewhere (perhaps on the Dota 2 bug you already opened).

If it contains `steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point` then it is probably this bug, or something closely related to this bug.

If it does not contain either of those then there is something weirder going on, possibly related to this bug.

**Comment by smcv** [comment]
@manticore-projects: The output you quoted is unfortunately not enough for me to tell whether you are seeing this bug, or something different. Please look a bit further back in the output from Steam, look for the same lines I described in my previous comment, and copy/paste that information.

**Comment by smcv** [comment]
> Hopefully, LIBTHAI_DICTDIR is properly setup before starting dota2 (and I suspect similar issue with ALSA_CONFIG_DIR and XLOCALEDIR)

When the sniper container runtime is used correctly, it runs the game in a container where `/usr` comes from the runtime and contains all of the necessary files, making it unnecessary (and potentially harmful) to set environment variables like these. The bug that we're tracking here is that the Steam client has updated Dota 2 to a version that expects to be run in the sniper container runtime, but it is not yet reliably using the sniper container runtime for that new version.

**Comment by smcv** [comment]
> after letting Steam update Dota 2, completely exit from Steam, then re-run it

Unfortunately this does not seem to be a 100% reliable workaround. On one of my test systems, this was enough to make Steam use the correct container runtime for Dota 2, but on a different test system, restarting Steam was not enough.

Forcing sniper to be installed by running `steam steam://install/1628350` might help (and certainly won't hurt), but the test system where I am still seeing this bug already had Sniper installed.

**Comment by smcv** [comment]
> In logs/compat_log.txt I have clearly:
> Requesting mapping AppID 570 from appinfo to tool "SteamLinuxRuntime_sniper"
> Mapping AppID 570 to tool "SteamLinuxRuntime_sniper" with priority 85

On the system where restarting Steam worked around this issue (Dota 2 correctly starts in sniper), I'm seeing the same thing as quoted here. But on the system where I can still reproduce this issue (Dota 2 incorrectly starts in steamlinuxruntime), in my `logs/compat_log.txt` I see something different:

```
Mapping AppID 570 to tool "steamlinuxruntime" with priority 250
...
Requesting mapping AppID 570 from appinfo to tool "SteamLinuxRuntime_sniper"
```

So I wonder whether there is a conflict happening between two different sources of information: a new record that says "Dota 2 should use SteamLinuxRuntime_sniper", and an older record that says "Dota 2 should use steamlinuxruntime".

Doing `app_info_print 570 4` in the console still gives me the correct information (with a reference to sniper) on both systems.

@sylware, it would probably be useful if you attach your whole `compat_log.txt` so we can compare. It's weird that you quoted the same thing from `compat_log.txt` as on my system that is working correctly, and yet when you launch the game, you get different behaviour.

**Comment by smcv** [comment]
And here's a system that exhibits this issue (it happens to be Debian 12, but that might not matter). Surprisingly, Dota 2 starts up OK for me even though it's in the wrong runtime, for whatever reason - but this is a situation where it "shouldn't" work correctly, and someone less lucky than me would probably see it fail. I didn't test whether gameplay was possible, only that it got far enough to show the red logo on a black background, and then pressed Alt+F4 and waited for it to exit.

In Steam's output:

`/bin/sh\0-c\0/home/desktop/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/desktop/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime'/scout-on-soldier-entry-point-v2 --  '/home/desktop/SteamLibrary/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0`
`Game process added : AppID 570 "/home/desktop/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/desktop/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime'/scout-on-soldier-entry-point-v2 --  '/home/desktop/SteamLibrary/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama", ProcID 4085, IP 0.0.0.0:0`

and some particularly interesting lines from `compat_log.txt`:

```
[2023-07-13 12:47:14] Mapping AppID 570 to tool "steamlinuxruntime" with priority 250
...
[2023-07-13 12:47:34] Requesting mapping AppID 570 from appinfo to tool "SteamLinuxRuntime_sniper"
...
[2023-07-13 12:48:15] StartSession: appID 570 session 651ef8c8b79f0809
[2023-07-13 12:48:15] Wildcard mapping tool matched with itself - ignoring: 1070560 steamlinuxruntime
[2023-07-13 12:48:18] Tool 1070560 "Steam Linux Runtime" has a dependency on tool 1391110.
[2023-07-13 12:48:18] Command prefix for tool 1391110 "Steam Linux Runtime - Soldier" set to: "'/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- ".
[2023-07-13 12:48:18] Command prefix for tool 1070560 "Steam Linux Runtime" set to: "'/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime'/scout-on-soldier-entry-point-v2 -- ".
```

Full logs

* [bug9835-compat_log.txt.gz](https://github.com/ValveSoftware/steam-for-linux/files/12039069/bug9835-compat_log.txt.gz)
* [bug9835-dota2-steam-output.txt.gz](https://github.com/ValveSoftware/steam-for-linux/files/12039070/bug9835-dota2-steam-output.txt.gz)

**Comment by sylware** [comment]
Nope, dota2 is not launch with (with your paths not mine): /home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point

**Comment by smcv** [comment]
> Nope, dota2 is not launch with (with your paths not mine): /home/desktop/SteamLibrary/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point

Please copy/paste your equivalent of the information in https://github.com/ValveSoftware/steam-for-linux/issues/9835#issuecomment-1634112037, and attach the logs. Just saying "not the same" is never going to be enough information for anyone to be able to diagnose why your system is behaving differently.

**Comment by sylware** [comment]
https://paste.c-net.org/SapphireSantos

There, dota2 is running ok with the symlinks and the libthai environment variable into the sniper steamruntime.

**Comment by smcv** [comment]
> `/bin/sh\0-c\0/mnt/128/home/user/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /mnt/128/home/user/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/mnt/128/home/user/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0`

This is not quite the same situation as what I reported: here, Dota 2 is not being run under a compat tool at all (not sniper as it should be now, but also not soldier + scout, which was the default for several months). Have you changed some setting in order to achieve this?

**Comment by manticore-projects** [comment]
```
are@ryzen ~> "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh"
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x5611ae2a8030
Using breakpad crash handler
[S_API] SteamAPI_Init(): Loaded '/home/are/.local/share/Steam/linux64/steamclient.so' OK.
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
07/13 20:40:26 Init: Starting.
07/13 20:40:26 InitHandler
07/13 20:40:26 Init Initializing.
07/13 20:40:26 Init: Installing breakpad exception handler for appid(570)/version(8188799)/tid(37065)
07/13 20:40:26 Init Done.
07/13 20:40:26 SetAppID Setting app ID to 570.
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198412125772 [API loaded yes]
SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561198412125772
07/13 20:40:26 Setting Steam ID: 76561198412125772
07/13 20:40:26 Setting SteamUniverse: 1 = Public
Setting breakpad minidump AppID = 373300
07/13 20:40:26 SetAppID Setting app ID to 373300.
Loaded libSDL2-2.0.so.0, got 0x5611ae570d20
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x5611ae2160b0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x5611ae5ce830
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x5611aeabc080
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x5611aedf9170
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x5611aedc1fe0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x5611aee00450
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x5611aee14e00
Fossilize INFO: Overriding serialization path: "/home/are/.local/share/Steam/steamapps/shadercache/570/fozpipelinesv6/steamapprun_pipeline_cache".
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libmaterialsystem2.so, got 0x5611b88c27e0
Loaded libpangoft2-1.0.so, got (nil)
 failed to dlopen "libpangoft2-1.0.so" error=/usr/lib/libharfbuzz.so.0: undefined symbol: FT_Get_Color_Glyph_Layer
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so, got (nil)
 failed to dlopen "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so" error=/usr/lib/libharfbuzz.so.0: undefined symbol: FT_Get_Color_Glyph_Layer
07/13 20:40:37 Breakpad_FilterCallback
07/13 20:40:37 FilterCallback Calling pre minidump callback
07/13 20:40:37 SetComment Setting comment to error

```

**Comment by smcv** [comment]
> On one of my test systems, this was enough to make Steam use the correct container runtime for Dota 2, but on a different test system, restarting Steam was not enough.

On the system where I was still seeing scout + soldier after a Steam restart, this was because I had overridden Dota 2 with: "Compatibility -> Force the use of a specific Steam Play compatibility tool -> Steam Linux Runtime" (which should have been equivalent to the default behaviour until now), and this override was not cleared when Dota 2 switched to requiring sniper. Clearing that override by unchecking "Force the use of a specific Steam Play compatibility tool" made it behave correctly.

In this situation, because "Steam Linux Runtime" (scout + soldier) is not compatible with sniper, I think ideally Steam would forcibly remove the override when a game switches to sniper as its required runtime.

**Comment by sylware** [comment]
Wait, I don't have proton enabled via the settings (like I don't have any shader cache), don't you tell me you cannot run dota2 with proton disabled???

**Comment by smcv** [comment]
> Wait, I don't have proton enabled via the settings (like I don't have any shader cache), don't you tell me you cannot run dota2 with proton disabled???

Dota 2 does not require Proton. It's a native Linux game.

However, it does rely on some of the same mechanisms that are used behind the scenes to run Proton.

**Comment by smcv** [comment]
On https://github.com/ValveSoftware/Dota-2/issues/2390, @manticore-projects wrote:
> This is so strange: Activating Proton worked for me and Dota is starting now.

with a screenshot of: global Steam Settings -> Compatibility -> Run other titles with -> set to Proton Experimental.

This is surprising to me, but maybe it will be the information that a Steam client developer needs to understand what has gone wrong.

If you had chosen to use Proton *for Dota 2 specifically* (in Properties -> Compatibility -> Force the use of ...), then that would mean that Steam would replace the native Linux version of Dota 2 with the Windows version, and run it under Proton. I would have understood it if *that* worked around this issue.

However, that's not what you've done in that screenshot! Instead, you've told Steam to use Proton Experimental for games that *don't* have a native Linux version. I don't know why that would change the Steam client's behaviour when running games that *do* have a native Linux version, like Dota 2.

@manticore-projects, some things that would probably be useful to know about this situation:

* What is the full command-line that Steam is using to run Dota 2 now - the equivalent of what I quoted in https://github.com/ValveSoftware/steam-for-linux/issues/9835#issuecomment-1634248032?

* If you disable this setting again (go back to how you had configured Steam before, whatever that is), does Dota 2 still work?

Your updated `~/.steam/steam/logs/compat_log.txt` would probably also be useful information for a Steam client developer to be able to debug this.

**Comment by smcv** [comment]
In https://github.com/ValveSoftware/Dota-2/issues/2390#issuecomment-1635176138, @jn64 wrote:
> Only way I found to launch Dota so far is by setting the compat to Steam Linux Runtime, which runs it with soldier (??)

That's not guaranteed to work, especially in the long term, but if it happens to work on your system then it's OK as a short term workaround. Please remember that you have changed this setting: after the underlying Steam client bug gets fixed, you will probably need to turn off "Force the use of ..." so that the game will run in sniper as it is meant to.

**Comment by manticore-projects** [comment]
@smcv: First of all, big thank you for helping! (Although I must say that all this stuff is extremely confusing on how it just happened and also how it was caused.)

Please find the log attached. It looks like I run on Sniper (who came up with that name in the context of Dota!) now. I have no idea, how and why since I only changed those two option. After changing, nothing extra was downloaded. Hope this helps.

```
are@ryzen ~> steam
steam.sh[87011]: Running Steam on arch rolling 64-bit
steam.sh[87011]: STEAM_RUNTIME is enabled automatically
setup.sh[87085]: Steam runtime environment up-to-date!
steam.sh[87011]: Steam client's requirements are satisfied
[2023-07-14 16:35:40] Startup - updater built Jul 10 2023 23:58:24
[2023-07-14 16:35:40] Startup - Steam Client launched with: '/home/are/.local/share/Steam/ubuntu12_32/steam'
07/14 16:35:40 Init: Installing breakpad exception handler for appid(steam)/version(1689034492)/tid(87142)
[2023-07-14 16:35:40] Loading cached metrics from disk (/home/are/.local/share/Steam/package/steam_client_metrics.bin)
[2023-07-14 16:35:40] Using the following download hosts for Public, Realm steamglobal
[2023-07-14 16:35:40] 1. https://client-update.akamai.steamstatic.com, /, Realm 'steamglobal', weight was 1000, source = 'update_hosts_cached.vdf'
[2023-07-14 16:35:40] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'update_hosts_cached.vdf'
[2023-07-14 16:35:40] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2023-07-14 16:35:40] Verifying installation...
[2023-07-14 16:35:40] Verification complete
Loaded SDL version 3.0.0-1782-g214d5daa3
XRRGetOutputInfo Workaround: initialized with override: 0 real: 0xec494db0
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0xec493500
GetWin32Stats: display was not open yet, good
ComputeStartupMode: found registry default startup mode: 0
Switching to desktopui, since -vgui was not specified
GetWin32Stats: display was not open yet, good
steamwebhelper.sh[87169]: Runtime for steamwebhelper: defaulting to /home/are/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
steamwebhelper.sh[87169]: glibc >= 2.34, partially disabling sandbox until CEF supports clone3()
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
CAppInfoCacheReadFromDiskThread took 16 milliseconds to initialize
Failed to init SteamVR because it isn't installed
local (potentially out of sync) copy of roaming config loaded - 2057 bytes.
BRefreshApplicationsInLibrary 1: 0ms
roaming config store loaded successfully - 2057 bytes.
migrating temporary roaming config store
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
System startup time: 19.19 seconds
BuildCompleteAppOverviewChange: 153 apps
RegisterForAppOverview 1: 7ms
RegisterForAppOverview 2: 7ms
ExecCommandLine: "'/home/are/.local/share/Steam/ubuntu12_32/steam'"
Fossilize INFO: Setting autogroup scheduling.
ExecuteSteamURL: "steam://open/steam://rungameid/570"
ExecuteSteamURL: "steam://rungameid/570"
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to UpdatingAppInfo with ""
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to ProcessingShaderCache with ""
GameAction [AppID 570, ActionID 1] : LaunchApp waiting for user response to ProcessingShaderCache ""
Fossilize INFO: Setting autogroup scheduling.
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 570, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 570, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
/bin/sh\0-c\0/home/are/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/are/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/are/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama\0
Game process added : AppID 570 "/home/are/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/are/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/are/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama", ProcID 87655, IP 0.0.0.0:0
chdir "/home/are/.local/share/Steam/steamapps/common/dota 2 beta"
ERROR: ld.so: object '/home/are/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/are/.local/share/Steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
ERROR: ld.so: object '/home/are/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/are/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/are/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
Could not connect to X session manager: Authentication Rejected, reason : MIT-MAGIC-COOKIE-1 authentication rejected
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""
GameAction [AppID 570, ActionID 1] : LaunchApp changed task to Completed with ""
pid 87758 != 87757, skipping destruction (fork without exec?)
pid 87759 != 87757, skipping destruction (fork without exec?)
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x55f456fe21e0
Using breakpad crash handler
[S_API] SteamAPI_Init(): Loaded '/home/are/.local/share/Steam/linux64/steamclient.so' OK.
Game process updated : AppID 570 "/home/are/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/are/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/are/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama", ProcID 87761, IP 0.0.0.0:0
Setting breakpad minidump AppID = 570
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
07/14 16:36:22 Init: Starting.
07/14 16:36:22 InitHandler
07/14 16:36:22 Init Initializing.
07/14 16:36:22 Init: Installing breakpad exception handler for appid(570)/version(8191169)/tid(87761)
07/14 16:36:22 Init Done.
07/14 16:36:22 SetAppID Setting app ID to 570.
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198412125772 [API loaded yes]
SteamInternal_SetMinidumpSteamID:  Setting Steam ID:  76561198412125772
07/14 16:36:22 Setting Steam ID: 76561198412125772
07/14 16:36:22 Setting SteamUniverse: 1 = Public
Setting breakpad minidump AppID = 373300
07/14 16:36:22 SetAppID Setting app ID to 373300.
Loaded libSDL2-2.0.so.0, got 0x55f4572a7bf0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libtier0.so, got 0x55f456f50a30
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libfilesystem_stdio.so, got 0x55f457031b40
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libinputsystem.so, got 0x55f457825fc0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/liblocalize.so, got 0x55f457b62ec0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/librendersystemvulkan.so, got 0x55f457b720a0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libresourcesystem.so, got 0x55f457b7e5d0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libschemasystem.so, got 0x55f457b58700
Fossilize INFO: Overriding serialization path: "/home/are/.local/share/Steam/steamapps/shadercache/570/fozpipelinesv6/steamapprun_pipeline_cache".
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libmaterialsystem2.so, got 0x55f461384e50
Loaded libpangoft2-1.0.so, got (nil)
 failed to dlopen "libpangoft2-1.0.so" error=libpangoft2-1.0.so: cannot open shared object file: No such file or directory
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so, got 0x55f45e5e5da0
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 38: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 39: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 40: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 41: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 42: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 43: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 44: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 45: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 46: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 47: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 48: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 49: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 50: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 51: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 52: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 53: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 54: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 55: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 56: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 57: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 58: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 59: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 60: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 61: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 62: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 63: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 64: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 65: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 66: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 67: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 68: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 69: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 70: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 71: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 72: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 73: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 74: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 75: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 76: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 77: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 78: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 79: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 80: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 81: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 82: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 83: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 84: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 85: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 86: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 87: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 88: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 89: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 90: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 91: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 92: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 93: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 94: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 95: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 96: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 97: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 98: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 99: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 100: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 101: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 102: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 103: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 104: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 105: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 106: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 107: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 108: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 109: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 110: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 111: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 112: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 113: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 114: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 115: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 116: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 117: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 118: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/fonts.conf", line 119: unknown element "fontpattern"
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/conf.d/41-repl-os-win.conf", line 148: Having multiple values in <test> isn't supported and may not work as expected
Fontconfig warning: "/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/core/panorama/fonts/conf.d/41-repl-os-win.conf", line 160: Having multiple values in <test> isn't supported and may not work as expected
Loaded libfreetype.so.6, got 0x55f45beefbb0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama_text_pango.so, got 0x55f45e5e5da0
Loaded libfontconfig.so.1, got 0x55f45f1cd020
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libmeshsystem.so, got 0x55f45cb7b270
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libworldrenderer.so, got 0x55f45f35e820
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libvscript.so, got 0x55f45d89ad70
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libnetworksystem.so, got 0x55f45de02800
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libanimationsystem.so, got 0x55f46188e240
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libvphysics2.so, got 0x55f4608268e0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libsoundsystem.so, got 0x55f460e876b0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libscenesystem.so, got 0x55f460fe47e0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libparticles.so, got 0x55f46174e6b0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota/bin/linuxsteamrt64/libserver.so, got 0x55f45ba41f90
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota/bin/linuxsteamrt64/libclient.so, got 0x55f45ec25ab0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libpanorama.so, got 0x55f46156d650
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libv8system.so, got 0x55f45d097d70
Loaded libmpg123.so.0, got 0x55f45e33c780
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libcairo.so, got 0x55f45d0577a0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libengine2.so, got 0x55f456fe21e0
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota/bin/linuxsteamrt64/libhost.so, got 0x55f45d592f80
Loaded /home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/bin/linuxsteamrt64/libscenefilecache.so, got 0x55f45dc3e3c0
CSteam3Client::Shutdown
Game process removed: AppID 570 "/home/are/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=570 -- /home/are/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/home/are/.local/share/Steam/steamapps/common/SteamLinuxRuntime_sniper'/_v2-entry-point --verb=waitforexitandrun --  '/home/are/.local/share/Steam/steamapps/common/dota 2 beta/game/dota.sh' +engine_experimental_drop_frame_ticks 1 +@panorama_min_comp_layer_dimension 0 -prewarm_panorama", ProcID 87761 
ThreadGetProcessExitCode: no such process 87763
ThreadGetProcessExitCode: no such process 87762
ThreadGetProcessExitCode: no such process 87761
ThreadGetProcessExitCode: no such process 87760
ThreadGetProcessExitCode: no such process 87757
ThreadGetProcessExitCode: no such process 87657
ThreadGetProcessExitCode: no such process 87656
Game 570 created interface STEAMAPPLIST_INTERFACE_VERSION001 / 
Game 570 created interface STEAMAPPS_INTERFACE_VERSION008 / 
Game 570 created interface STEAMHTMLSURFACE_INTERFACE_VERSION_005 / 
Game 570 created interface STEAMHTTP_INTERFACE_VERSION003 / 
Game 570 created interface STEAMHTTP_INTERFACE_VERSION003 / HTTP
Game 570 created interface STEAMINVENTORY_INTERFACE_V003 / 
Game 570 created interface STEAMMUSICREMOTE_INTERFACE_VERSION001 / 
Game 570 created interface STEAMMUSIC_INTERFACE_VERSION001 / 
Game 570 created interface STEAMPARENTALSETTINGS_INTERFACE_VERSION001 / 
Game 570 created interface STEAMREMOTESTORAGE_INTERFACE_VERSION016 / 
Game 570 created interface STEAMSCREENSHOTS_INTERFACE_VERSION003 / 
Game 570 created interface STEAMUGC_INTERFACE_VERSION016 / 
Game 570 created interface STEAMUSERSTATS_INTERFACE_VERSION012 / 
Game 570 created interface STEAMVIDEO_INTERFACE_V002 / 
Game 570 created interface SteamAppDisableUpdate001 / 
Game 570 created interface SteamController008 / 
Game 570 created interface SteamFriends017 / 
Game 570 created interface SteamGameCoordinator001 / 
Game 570 created interface SteamInput006 / 
Game 570 created interface SteamMatchGameSearch001 / 
Game 570 created interface SteamMatchMaking009 / 
Game 570 created interface SteamMatchMakingServers002 / 
Game 570 created interface SteamNetworking006 / 
Game 570 created interface SteamNetworkingSocketsSerialized005 / 
Game 570 created interface SteamUser021 / 
Game 570 created interface SteamUser021 / User
Game 570 created interface SteamUser022 / User
Game 570 created interface SteamUtils010 / 
Game 570 created interface SteamUtils010 / Utils
Game 570 method call count for IClientNetworkingSocketsSerialized::GetSTUNServer : 2
Game 570 method call count for IClientNetworkingSocketsSerialized::GetCachedRelayTicket : 1
Game 570 method call count for IClientNetworkingSocketsSerialized::GetCachedRelayTicketCount : 1
Game 570 method call count for IClientNetworkingSocketsSerialized::GetCertAsync : 1
Game 570 method call count for IClientAppDisableUpdate::SetAppUpdateDisabledSecondsRemaining : 1
Game 570 method call count for IClientUGC::GetSerializedQueryUGCResponse : 6
Game 570 method call count for IClientUGC::SetReturnLongDescription : 8
Game 570 method call count for IClientUGC::SetReturnKeyValueTags : 8
Game 570 method call count for IClientUGC::ReleaseQueryUGCRequest : 8
Game 570 method call count for IClientUGC::SendQueryUGCRequest : 8
Game 570 method call count for IClientUGC::CreateQueryUserUGCRequest : 8
Game 570 method call count for IClientHTTP::ReleaseHTTPRequest : 12
Game 570 method call count for IClientHTTP::GetHTTPResponseBodyData : 13
Game 570 method call count for IClientHTTP::GetHTTPResponseBodySize : 11
Game 570 method call count for IClientHTTP::GetHTTPResponseHeaderValue : 9
Game 570 method call count for IClientHTTP::GetHTTPResponseHeaderSize : 9
Game 570 method call count for IClientHTTP::SendHTTPRequest : 13
Game 570 method call count for IClientHTTP::SetHTTPRequestGetOrPostParameter : 1
Game 570 method call count for IClientHTTP::SetHTTPRequestHeaderValue : 1
Game 570 method call count for IClientHTTP::SetHTTPRequestNetworkActivityTimeout : 2
Game 570 method call count for IClientHTTP::SetHTTPRequestContextValue : 8
Game 570 method call count for IClientHTTP::CreateHTTPRequest : 13
Game 570 method call count for IClientGameCoordinator::RetrieveMessage : 40
Game 570 method call count for IClientGameCoordinator::IsMessageAvailable : 81
Game 570 method call count for IClientGameCoordinator::SendMessage : 32
Game 570 method call count for IClientRemoteStorage::GetFileNameAndSize : 38
Game 570 method call count for IClientRemoteStorage::GetFileCount : 2
Game 570 method call count for IClientRemoteStorage::FileExists : 28
Game 570 method call count for IClientRemoteStorage::GetFileSize : 22
Game 570 method call count for IClientRemoteStorage::FileWriteAsync : 9
Game 570 method call count for IClientRemoteStorage::FileRead : 21
Game 570 method call count for IClientNetworking::GetMaxPacketSize : 1
Game 570 method call count for IClientUserStats::GetNumberOfCurrentPlayers : 1
Game 570 method call count for IClientUserStats::RequestCurrentStats : 1
Game 570 method call count for IClientAppManager::GetCurrentLanguage : 1
Game 570 method call count for IClientAppManager::GetAppStateInfo : 1
Game 570 method call count for IClientUtils::RecordSteamInterfaceCreation : 118
Game 570 method call count for IClientUtils::FilterText : 70
Game 570 method call count for IClientUtils::InitFilterText : 1
Game 570 method call count for IClientUtils::IsSteamChina : 1
Game 570 method call count for IClientUtils::SetOverlayNotificationInset : 2
Game 570 method call count for IClientUtils::SetOverlayNotificationPosition : 3
Game 570 method call count for IClientUtils::GetLauncherType : 1
Game 570 method call count for IClientUtils::GetSteamUILanguage : 1
Game 570 method call count for IClientUtils::GetAPICallResult : 25
Game 570 method call count for IClientUtils::GetAppID : 125
Game 570 method call count for IClientUtils::GetImageRGBA : 23
Game 570 method call count for IClientUtils::GetImageSize : 23
Game 570 method call count for IClientUtils::GetIPCountry : 1
Game 570 method call count for IClientUtils::GetServerRealTime : 35
Game 570 method call count for IClientFriends::GetPlayerNickname_Public : 71
Game 570 method call count for IClientFriends::GetFriendPersonaName_Public : 14
Game 570 method call count for IClientFriends::SetListenForFriendsMessages : 3
Game 570 method call count for IClientFriends::GetFriendRichPresence : 112
Game 570 method call count for IClientFriends::SetRichPresence : 20
Game 570 method call count for IClientFriends::GetFriendGamePlayed : 11
Game 570 method call count for IClientFriends::GetFriendByIndex : 27
Game 570 method call count for IClientFriends::GetFriendCount : 3
Game 570 method call count for IClientFriends::GetLargeFriendAvatar : 3
Game 570 method call count for IClientFriends::GetMediumFriendAvatar : 29
Game 570 method call count for IClientFriends::GetSmallFriendAvatar : 29
Game 570 method call count for IClientFriends::GetFriendPersonaState : 11
Game 570 method call count for IClientFriends::GetFriendRelationship : 11
Game 570 method call count for IClientFriends::GetPersonaState : 146
Game 570 method call count for IClientFriends::GetPersonaName : 1
Game 570 method call count for IClientUser::GetDurationControl : 1
Game 570 method call count for IClientUser::GetUserDataFolder : 1
Game 570 method call count for IClientUser::GetSteamID : 35
Game 570 method call count for IClientUser::BLoggedOn : 4
Uploaded AppInterfaceStats to Steam
[2023-07-14 16:37:16] Shutdown
```

**Comment by manticore-projects** [comment]
Disabling the Compatibility Options leads to the same crash as before.
Or in other words, Setting those Options is mandatory or else no Dota.

**Comment by smcv** [comment]
> It looks like I run on Sniper now

OK, then with your workaround of enabling Proton for Windows-only games (!?), Dota 2 is working for you the way it is intended to work. That's progress!

I don't know why that workaround would be necessary or successful for you, but I don't have access to Steam client source code - I can only hope that a developer who does have that access will see this information and think "aha! now I understand what's happening", and use that as a basis for a proper fix.

> Disabling the Compatibility Options leads to the same crash as before.

To confirm, when you say "disabling the Compatibility Options", I think you mean: Steam menu -> Settings -> Compatibility -> disable (uncheck) "Enable Steam Play for all other titles", leaving "Enable Steam Play for supported titles" activated (checked). Is that true?

Or if that's not what you did, then what? Please be specific, small details can matter here.

When you have done this, does it take you back to the same command-line that I quoted in https://github.com/ValveSoftware/steam-for-linux/issues/9835#issuecomment-1634248032? If different, then please copy/paste what you do get - just that one line starting with `/bin/sh\0-c\0` is probably enough, we probably don't need all of the other output.

> (who came up with that name in the context of Dota!) 

The Steam Runtime is a general Steam-on-Linux feature and is used (in different forms) by all games, not just Dota 2. Its branches have always used TF2 characters as codenames. So far we have had scout, heavy, soldier and sniper; spy was an internal prototype that never got released, and medic is another internal prototype for a possible future version. There are not so many TF2 characters left, so the naming convention for versions beyond medic is likely to change to the boring but practical `steamrt5` and so on.
